src/core/base.js

```diff
@@ -75,6 +75,10 @@ getJasmineRequireObj().base = function(j$, jasmineGlobal) {
     return obj.nodeType > 0;
   };
 
+  j$.isPromise = function(obj) {
+    return (typeof Promise !== 'undefined') && obj.constructor === Promise;
```

Comment:
Would a more portable way to do this be `obj.constructor.name === 'Promise'`?  Line 79 as written would only support native `Promise`s.  I'm not sure it's necessary yet, but have you considered allowing users to tell Jasmine which Promise library you're using via something like `jasmine.setPromiseLib(require('bluebird'))` or similar?

Reply:
After talking this over with @slackersoft I'm going to go ahead and merge this PR. The drawback to checking for `then` is that it might incorrectly treat a non-promise object that has a `then` method as a promise. It would be nice to be able to extend this behavior to promise libraries like Bluebird, and I'd be happy to review a pull request that does that. But even if we only handle native promises, that's an improvement.